src/metasploit/msfrpc.py

```diff
@@ -311,9 +312,15 @@ def login(self, username, password):
         """
         if self.sessionid is None:
             r = self.call(MsfRpcMethod.AuthLogin, username, password)
+            # in case r is actually encoded in bytes, this is a safe way to turn it back to string for the try/catch
+            str_data = {
+                key.decode() if isinstance(key, bytes) else key:
+                    val.decode() if isinstance(val, bytes) else val
+                for key, val in r.items()
+            }
```

Comment:
Little difficult to read - splitting this up may help

Reply:
Personally I think it's readable enough, but could use a comment explaining verbally what's going on and why.

Reply:
I changed it to be longer/less efficient but more readable by a programmer